src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKey.Fixer.cs

```diff
@@ -0,0 +1,86 @@
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            var diagnostic = context.Diagnostics.FirstOrDefault();
+
+            if (TryParseLocationInfo(diagnostic, DoNotGuardDictionaryRemoveByContainsKey.ConditionalOperation, out var conditionalOperationSpan) &&
+                TryParseLocationInfo(diagnostic, DoNotGuardDictionaryRemoveByContainsKey.ChildStatementOperation, out var childStatementOperationSpan) &&
+                root.FindNode(conditionalOperationSpan) is SyntaxNode conditionalOperationNode &&
+                root.FindNode(childStatementOperationSpan) is SyntaxNode childStatementOperationNode)
+            {
+                context.RegisterCodeFix(new DoNotGuardDictionaryRemoveByContainsKeyCodeAction(_ =>
+                    Task.FromResult(ReplaceConditionWithChild(context.Document, root, conditionalOperationNode, childStatementOperationNode))),
+                    diagnostic);
+            }
+        }
+
+        private static Document ReplaceConditionWithChild(Document document, SyntaxNode root, SyntaxNode conditionalOperationNode, SyntaxNode childOperationNode)
+        {
+            var newNode = childOperationNode.WithAdditionalAnnotations(Formatter.Annotation);
+            var newRoot = root.ReplaceNode(conditionalOperationNode, newNode);
+
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private static bool TryParseLocationInfo(Diagnostic diagnostic, string propertyKey, out TextSpan span)
+        {
+            span = default;
+
+            if (!diagnostic.Properties.TryGetValue(propertyKey, out var locationInfo))
+                return false;
+
+            var parts = locationInfo.Split(new[] { DoNotGuardDictionaryRemoveByContainsKey.AdditionalDocumentLocationInfoSeparator }, StringSplitOptions.None);
+            if (parts.Length != 2 ||
+                !int.TryParse(parts[0], out var spanStart) ||
+                !int.TryParse(parts[1], out var spanLength))
+            {
+                return false;
+            }
+
+            span = new TextSpan(spanStart, spanLength);
+            return true;
+        }
+
+        private class DoNotGuardDictionaryRemoveByContainsKeyCodeAction : DocumentChangeAction
+        {
+            public DoNotGuardDictionaryRemoveByContainsKeyCodeAction(Func<CancellationToken, Task<Document>> action)
+            : base(MicrosoftNetCoreAnalyzersResources.DoNotGuardDictionaryRemoveByContainsKeyTitle, action,
```

Comment:
Codefix title should be different from issue title as it is supposed to give the action that will be executed.